### PR TITLE
Option to manually invalidate reader to drop packets

### PR DESCRIPTION
**Pull request by euttsov-odaq**
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description:

### CODE REVIEW

File: core/opendaq/reader/include/opendaq/multi_reader.h [hunk 1]
```diff
@@ -156,6 +156,9 @@ DECLARE_OPENDAQ_INTERFACE(IMultiReader, ISampleReader)
      * @param commonSampleRate The domain point at which the reader managed to synchronize all the signals.
      */
     virtual ErrCode INTERFACE_FUNC getCommonSampleRate(Int* commonSampleRate) = 0;
+
+    virtual ErrCode INTERFACE_FUNC setActive(Bool isActive) = 0;
```
Comment (denise-opendaq) [inline core/opendaq/reader/include/opendaq/multi_reader.h:160]: comments are missing

File: core/opendaq/reader/src/multi_reader_impl.cpp [hunk 2]
```diff
@@ -311,7 +312,7 @@ ListPtr<IInputPortConfig> MultiReaderImpl::CheckPreconditions(const ListPtr<ICom
                 throw InvalidParameterException("Cannot pass both input ports and signals as items");
 
             if (overrideMethod && port.getConnection().assigned())
-                throw InvalidParameterException("Signal has to be connected to port after reader is created");
+                throw InvalidParameterException("Signal has to be connected to the port before the reader is created");
```
Comment (JakaMohorkoDS) [inline core/opendaq/reader/src/multi_reader_impl.cpp:315]: Isn't this the opposite? The signal must not yet be connected before a the reader is created.

File: core/opendaq/reader/src/multi_reader_impl.cpp [hunk 3]
```diff
@@ -863,7 +864,7 @@ ErrCode MultiReaderImpl::connected(IInputPort* port)
     {
         for (auto& signal: signals)
         {
-            signal.port.setActive(true);
+            signal.port.setActive(true && isActive);
```
Comment (JakaMohorkoDS) [inline core/opendaq/reader/src/multi_reader_impl.cpp:867]: maybe just `setActive(isActive)` :D

**Review by JakaMohorkoDS — COMMENTED** [review]
Lgtm. Tests could include a standard use case of how one would make the reader inactive from within the `dataAvailable` callback. Additionally, a test where multiple events are enqueued, with data packets interleaved is in place - do we correctly drop all the packets until the next event, and notify the user about the last event in the queue?

Also - the power and fft reference function blocks should be modified to make use of the `setActive` reader  call when users correct unaccepted signals.

**Review by JakaMohorkoDS — APPROVED** [review]

**Comment by JakaMohorkoDS** [comment]
Lgtm!

**Review by dejancrn — APPROVED** [review]

**Comment by euttsov-odaq** [comment]
> Lgtm. Tests could include a standard use case of how one would make the reader inactive from within the `dataAvailable` callback. Additionally, a test where multiple events are enqueued, with data packets interleaved is in place - do we correctly drop all the packets until the next event, and notify the user about the last event in the queue?
> 
> Also - the power and fft reference function blocks should be modified to make use of the `setActive` reader call when users correct unaccepted signals.

Added test for onDataAvailable callback

**Review by JakaMohorkoDS — APPROVED** [review]

**Pull request merged**